opm/parser/eclipse/EclipseState/Grid/GridDims.hpp

```diff
@@ -0,0 +1,174 @@
+/*
+  Copyright 2016  Statoil ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef OPM_PARSER_GRIDDIMS_HPP
+#define OPM_PARSER_GRIDDIMS_HPP
+
+#include <array>
+#include <stdexcept>
+#include <vector>
+
+#include <opm/parser/eclipse/Deck/Deck.hpp>
+#include <opm/parser/eclipse/Deck/DeckKeyword.hpp>
+#include <opm/parser/eclipse/Deck/DeckRecord.hpp>
+#include <opm/parser/eclipse/Deck/Section.hpp>
+
+#include <opm/parser/eclipse/Parser/MessageContainer.hpp>
+#include <opm/parser/eclipse/Parser/ParserKeywords/D.hpp> // DIMENS
+#include <opm/parser/eclipse/Parser/ParserKeywords/S.hpp> // SPECGRID
+
+namespace Opm {
+    class GridDims
+    {
+    public:
+
+        GridDims(std::array<int, 3> xyz) :
+            GridDims(xyz[0], xyz[1], xyz[2])
+        {
+        }
+
+        GridDims(size_t nx, size_t ny, size_t nz) :
+            m_nx(nx), m_ny(ny), m_nz(nz)
+        {
+        }
+
+        GridDims(const Deck& deck, MessageContainer messages = MessageContainer())
+        {
+            std::vector<int> xyz;
```

Comment:
This should in my opinion be simplified massively:
```
private void GridDims::init(const DeckKeyword& keyword) {
   m_nx = keyword.getItem("NX").getInt(0);
   m_ny = keyword.getItem("NY").getInt(0);
   m_nz = keyword.getItem("NZ").getInt(0);
}

GridDims::GridDims(const Deck& deck) {
   if (deck.hasKeyword("SPECGRID"))
      init( deck.getKeyword("SPECGRID"))
   else if (deck.hasKeyword("DIMENS"))
      init( deck.getKeyword("DIMENS"));
   else
      throw std::invalid_argument("Must have either SPECGRID or DIMENS to indicate grid dimesnions");
}
```
This implementation exploits the fact that the `SPECGRID` and `DIMENS` share the `NX,NY,NZ` part of the configuration - if going that way we should probably add a test to enforce that behavior.